lib/alex_koin/slack.ex

```diff
@@ -112,15 +137,22 @@ defmodule AlexKoin.SlackRtm do
     |> @slack_module.send_raw(slack)
   end
 
-  defp reason(text) do
-    %{"reason" => reason} = Regex.named_captures(~r/create koin\s+(?<reason>.*)/, text)
-    reason
+  defp notify_creator(creator, reason, slack) do
+    msg = "You just mined an :akc: for '#{reason}'."
+    dm_channel = dm_channel_for_slack_id(creator.slack_id, slack.ims)
+
+    if dm_channel do
```

Comment:
There's a lot of send_raw_message after checking for dm_channel. I believe the "Elixir way" would be to make a function for these that pattern match on the input.

```
defp send_dm_message(channel, options)
  # options has slack, msg, etc
  # send_raw_message things
end
defp send_dm_message(nil, options)
  # Handle the nil case
end
```

You could probably even make just one with a `when` clause, but I think if statements are pretty faux pas in Elixir.